exchange/bitswap/wantmanager.go

```diff
@@ -168,11 +168,17 @@ func (mq *msgQueue) runQueue(ctx context.Context) {
 }
 
 func (pm *WantManager) Connected(p peer.ID) {
-	pm.connect <- p
+	select {
+	case pm.connect <- p:
+	case <-pm.ctx.Done():
+	}
 }
```

Comment:
These seem like odd, little helpers.  Trying to understand their utility, it looks like `WantManager.Connected` is only used by `Bitswap.PeerConnected`, which is in turn used by some bitswap tests and `netNotifiee.Connected`.  But I can't figure out where these tie into exchange interface.  Can you give me a few pointers?